test/cat.jl

```diff
@@ -88,27 +88,18 @@ module TestCat
     dt[1] = zeros(Int, nrow(dt))
     vcat(dt, alt_dt)
 
-    # Don't fail on non-matching names
-    names!(alt_dt, [:A, :B, :C])
-    vcat(dt, alt_dt)
-
     dtr = vcat(dt4, dt4)
     @test size(dtr, 1) == 8
     @test names(dt4) == names(dtr)
     @test isequal(dtr, [dt4; dt4])
 
-    dtr = vcat(dt2, dt3)
```

Comment:
Ah, and the behavior should also be documented in the docstrings. IIUC, `vcat` now follows the behavior of SQL `UNION ALL`, i.e. both names and order of columns must match? The error messages should also mention the order then (and which columns are in different positions).